### DISCUSSION TITLE
When i use dpdata and cp2kdata to convert cp2k MD data,I encountered an error,The content of this error is :dpdata.data_type.DataError: Shape of coords is (67, 1056, 3), but expected (68, 1056, 3).

### DISCUSSION THREAD

**Discussion by llllkkkkkoo** [deepmodeling/dpdata#622] [Q&A] [answered]
When i use dpdata and cp2kdata to convert cp2k MD data,I encountered an error,The error message is as follows:
--- You are parsing data using package Cp2kData ---
Parsing Energies from aimd/Li-LiFSI-DME-1-2-1.ener
Parsing Structures from aimd/Li-LiFSI-DME-1-2-pos-1.xyz
Parsing Froces from aimd/Li-LiFSI-DME-1-2-frc-1.xyz
Parsing Stress from the CP2K output/log file: aimd/output
Parsing Cells Information from aimd/output
Atom names are fake chemical symbols as you set in cp2k input.
--- You are parsing data using package Cp2kData ---
Traceback (most recent call last):
  File "./readdata_cp2k.py", line 6, in <module>
    data = dpdata.LabeledSystem('aimd',cp2k_output_name='output', fmt = 'cp2kdata/md')
  File "/fs2/home/chengroup/.local/lib/python3.8/site-packages/dpdata/system.py", line 183, in __init__
    self.from_fmt(
  File "/fs2/home/chengroup/.local/lib/python3.8/site-packages/dpdata/system.py", line 220, in from_fmt
    return self.from_fmt_obj(load_format(fmt), file_name, **kwargs)
  File "/fs2/home/chengroup/.local/lib/python3.8/site-packages/dpdata/system.py", line 1120, in from_fmt_obj
    self.check_data()
  File "/fs2/home/chengroup/.local/lib/python3.8/site-packages/dpdata/system.py", line 207, in check_data
    dd.check(self)
  File "/fs2/home/chengroup/.local/lib/python3.8/site-packages/dpdata/data_type.py", line 112, in check
    raise DataError(
dpdata.data_type.DataError: Shape of coords is (67, 1056, 3), but expected (68, 1056, 3)
I hope to get your help,thank you very much.
[files.zip](https://github.com/deepmodeling/dpdata/files/14740480/files.zip)

**Comment by robinzyb** [comment]
could you include `1-2-1.ener` `aimd/output` in the zip as well?

**Reply by llllkkkkkoo** [reply]
yes,the energy and foece is output every ten steps

**Reply by llllkkkkkoo** [reply]
I have scrutinized my ener file 、my pos-1.xyz file 、 my frc-1.xyz file and my out file, they contain the same configurations.I think the reason is that the out file contain the cell information of step 0,but xyz files do not contain the information of step 0.

**Reply by llllkkkkkoo** [reply]
this is a complete 1-2-1.ener  file
[1-2-1ener.zip](https://github.com/deepmodeling/dpdata/files/14742988/1-2-1ener.zip)

**Reply by robinzyb** [reply]
> files do not contain the information of step 

probably. Is this a restart md simulation? if so, please send me the files so that I can improve cp2kdata. Also, I found the frequency of energies output is different from the frequency for pos.

**Reply by llllkkkkkoo** [reply]
yes,it is a restart md simulation,do I send you my reatart files? I use a python script for my ener files to ensure the output frequency of my ener file is same as my pos file and frc file.

**Reply by llllkkkkkoo** [reply]
this is my output files ,I hope that will be helpful for you to improve cp2kdata.
[files.zip](https://github.com/deepmodeling/dpdata/files/14744818/files.zip)

**Reply by robinzyb** [reply]
thanks. I will fix it asap

**Reply by llllkkkkkoo** [reply]
thanks a lot.

**Reply by llllkkkkkoo** [reply]
Could you please tell me if you have updated the cp2kdata package?

**Comment by llllkkkkkoo** [comment]
Could you please tell me if  you have updated the cp2kdata package?

**Reply by robinzyb** [reply]
not yet, I will update today. I am developing other codes for now

**Reply by robinzyb** [reply]
I am almost finishing the debugging. Could you update a smaller restart md outputs ,for example, a restart md with onlyt 2-3 steps, includes `-pos-`, `-frc-`, `ener`, and `output`?  
That would be a great help!
I will include them into cp2kdata testsuits.

**Reply by robinzyb** [reply]
I found no restart information in your cp2k output file. I guess you directly submit the `Li-LiFSI-DME-1-2-1.restart` through `cp2k.psmp -i Li-LiFSI-DME-1-2-1.restart `, which is not recommened by cp2k.
An alternative way to submit restart is using 
```
&EXT_RESTART
      RESTART_FILE_NAME Li-LiFSI-DME-1-2-1.restart
&END EXT_RESTART 
```

This will write header for restart  in cp2k output

```
 *******************************************************************************
 *                            RESTART INFORMATION                              *
 *******************************************************************************
 *                                                                             *
 *    RESTART FILE NAME: bivo4-water-1.restart                                 *
 *                                                                             *
 * RESTARTED QUANTITIES:                                                       *
 *                       CELL                                                  *
 *                       COORDINATES                                           *
 *                       RANDOM NUMBER GENERATOR                               *
 *                       VELOCITIES                                            *
 *                       MD COUNTERS                                           *
 *                       MD AVERAGES                                           *
 *                       PARTICLE THERMOSTAT                                   *
 *                       REAL TIME PROPAGATION                                 *
 *                       PINT BEAD POSITIONS                                   *
 *                       PINT BEAD VELOCITIES                                  *
 *                       PINT NOSE THERMOSTAT                                  *
 *                       PINT GLE THERMOSTAT                                   *
 *                       HELIUM BEAD POSITIONS                                 *
 *                       HELIUM PERMUTATION STATE                              *
 *                       HELIUM FORCES ON SOLUTE                               *
 *                       HELIUM RNG STATE                                      *
 *******************************************************************************
```
With this header, I can let cp2kdata automatically detect whether it is a restart output file.
And I will write an extra option for telling cp2kdata if it is a restart output.

**Reply by llllkkkkkoo** [reply]
i always use restart file of cp2k to continue calculate,I really didn't notice that the output file do not include the restart information,
because it outputs a separate file called .restart file,so i do not allow the restart information to output the output file.

**Reply by llllkkkkkoo** [reply]
I guess you directly submit the Li-LiFSI-DME-1-2-1.restart through cp2k.psmp -i Li-LiFSI-DME-1-2-1.restart , which is not recommened by cp2k.
That's exactly what I did.
Thank you for your suggestion,i will test this method.

**Comment by robinzyb** [comment]
Just in case of your future calculations, please follow the recommended cp2k setups in https://github.com/robinzyb/cp2kdata/blob/main/docs/dpdata_plugin.md

**Reply by llllkkkkkoo** [reply]
i indeed reference this standard,but i do not notice this setting about restarting.

**Comment by llllkkkkkoo** [comment]
> I found no restart information in your cp2k output file. I guess you directly submit the `Li-LiFSI-DME-1-2-1.restart` through `cp2k.psmp -i Li-LiFSI-DME-1-2-1.restart `, which is not recommened by cp2k. An alternative way to submit restart is using
> 
> ```
> &EXT_RESTART
>       RESTART_FILE_NAME Li-LiFSI-DME-1-2-1.restart
> &END EXT_RESTART 
> ```
> 
> This will write header for restart in cp2k output
> 
> ```
>  *******************************************************************************
>  *                            RESTART INFORMATION                              *
>  *******************************************************************************
>  *                                                                             *
>  *    RESTART FILE NAME: bivo4-water-1.restart                                 *
>  *                                                                             *
>  * RESTARTED QUANTITIES:                                                       *
>  *                       CELL                                                  *
>  *                       COORDINATES                                           *
>  *                       RANDOM NUMBER GENERATOR                               *
>  *                       VELOCITIES                                            *
>  *                       MD COUNTERS                                           *
>  *                       MD AVERAGES                                           *
>  *                       PARTICLE THERMOSTAT                                   *
>  *                       REAL TIME PROPAGATION                                 *
>  *                       PINT BEAD POSITIONS                                   *
>  *                       PINT BEAD VELOCITIES                                  *
>  *                       PINT NOSE THERMOSTAT                                  *
>  *                       PINT GLE THERMOSTAT                                   *
>  *                       HELIUM BEAD POSITIONS                                 *
>  *                       HELIUM PERMUTATION STATE                              *
>  *                       HELIUM FORCES ON SOLUTE                               *
>  *                       HELIUM RNG STATE                                      *
>  *******************************************************************************
> ```
> 
> With this header, I can let cp2kdata automatically detect whether it is a restart output file. And I will write an extra option for telling cp2kdata if it is a restart output.

thank you very much,i will test cp2kdata again.

**Comment by llllkkkkkoo** [comment]
I have one more question,now if all versions of cp2kdata have updatad?

**Reply by robinzyb** [reply]
> I am almost finishing the debugging. Could you update a smaller restart md outputs ,for example, a restart md with onlyt 2-3 steps, includes `-pos-`, `-frc-`, `ener`, and `output`? That would be a great help! I will include them into cp2kdata testsuits.

could you update smaller md files (with only 2-3 steps), so that I can test and update the code.

**Reply by llllkkkkkoo** [reply]
I'll send it to you as soon as possible.

**Comment by llllkkkkkoo** [comment]
> > I am almost finishing the debugging. Could you update a smaller restart md outputs ,for example, a restart md with onlyt 2-3 steps, includes `-pos-`, `-frc-`, `ener`, and `output`? That would be a great help! I will include them into cp2kdata testsuits.
> 
> could you update smaller md files (with only 2-3 steps), so that I can test and update the code.

these are my all output files,i hope these files are helpful to you.
[cp2k.zip](https://github.com/deepmodeling/dpdata/files/14942278/cp2k.zip)

**Reply by robinzyb** [reply, marked as answer]
New version of cp2kdata `0.6.7` is released.
you can update your cp2kdata by 
`pip install cp2kdata -U`
Your cp2k md files are included in the testsuit of cp2kdata in 
[`cp2kdata/tests/test_dpdata/v2024.1/aimd_npt_i_restart`](https://github.com/robinzyb/cp2kdata/tree/b8c241b4064dd82e8508f5ac8ab98d4bf72d49f6/tests/test_dpdata/v2024.1/aimd_npt_i_restart)
The documentation for parsing restarted md is updated in 
https://robinzyb.github.io/cp2kdata/docs/dpdata_plugin.html

Thank you for your help.

**Comment by llllkkkkkoo** [comment]
> New version of cp2kdata `0.6.7` is released. you can update your cp2kdata by `pip install cp2kdata -U` Your cp2k md files are included in the testsuit of cp2kdata in [`cp2kdata/tests/test_dpdata/v2024.1/aimd_npt_i_restart`](https://github.com/robinzyb/cp2kdata/tree/b8c241b4064dd82e8508f5ac8ab98d4bf72d49f6/tests/test_dpdata/v2024.1/aimd_npt_i_restart) The documentation for parsing restarted md is updated in https://robinzyb.github.io/cp2kdata/docs/dpdata_plugin.html
> 
> Thank you for your help.

thank you.
